pkg/cmd/server/server.go

```diff
@@ -196,13 +216,16 @@ func (c *Config) Complete(ctx context.Context) (RunnableServer, error) {
 
 	var cachingClusterDispatch dispatch.Dispatcher
 	if c.DispatchServer.Enabled {
-		cdcc, cerr := c.ClusterDispatchCacheConfig.Complete()
-		if cerr != nil {
-			return nil, fmt.Errorf("failed to configure cluster dispatch: %w", cerr)
+		cdcc, err := c.ClusterDispatchCacheConfig.Complete()
+		if err != nil {
+			return nil, fmt.Errorf("failed to configure cluster dispatch: %w", err)
 		}
 		log.Info().EmbedObject(cdcc).Msg("configured cluster dispatch cache")
+		closers = append(closers, func() error {
+			cdcc.Close()
+			return nil
+		})
```

Comment:
may as well just use defer here since this Close() cannot fail

Reply:
If we used defer, we would close it when `Complete` returns, which we don't want. We want to close it _if an error happens_. That's what the first defer statement that I've added does.